imxrt-hal/src/srtc.rs

```diff
@@ -117,27 +118,29 @@ fn is_enabled(snvs: &Instance) -> bool {
     ral::read_reg!(ral::snvs, snvs, LPCR, SRTC_ENV) != 0
 }
 
-/// Converts the number of milliseconds that have occurred since a second into clock ticks
+/// Converts the number of microseconds that have occurred since a second into clock ticks
 /// (1/32768 of a second).
 ///
-/// For example: for the time 2020-10-05 01:39:56.505, `millis` is `505`.
-pub fn subsec_millis_to_ticks(millis: u16) -> u16 {
-    // 999 is the max valid value outside of a leap second; clamp
-    let millis = millis.min(999);
-    let millis = millis as f32 / 1000.0;
-    let ticks = millis * 32768.0;
+/// For example: for the time `2020-10-05 01:39:56.505`, `micros` is `505000`, and this gives
+/// `16547` ticks.
+pub fn subsec_micros_to_ticks(micros: u32) -> u16 {
+    const RATIO: f32 = 0.032768; // 32768 / 1_000_000, each microsecond is this many ticks
+    let micros = micros.min(999_999);
+    // 999999 is the max valid value outside of a leap second; clamp
+    let ticks = micros as f32 * RATIO;
     ticks as u16
 }
 
 // these won't round trip in part due to the lack of round in no_std
 
-/// Converts clock ticks (1/32768 of a second) into milliseconds.
+/// Converts sub-second clock ticks (1/32768 of a second) into microseconds.
 ///
-/// For example: 32000 ticks works out to 976 milliseconds.
-pub fn ticks_to_subsec_millis(ticks: u16) -> u16 {
+/// For example: 32000 ticks works out to 976562 microseconds.
+pub fn ticks_to_subsec_micros(ticks: u16) -> u32 {
```

Comment:
The same math can be done above but with (ticks*NANOS_PER_TICK)/1000 to get microseconds

Reply:
The same commentary about making it public, if its public I think ticks_to_micros is good enough, that's what the math is doing here

Reply:
Changed the math and made it private, which also led me to remove the 15-bit mask as we now know the input is always going to be 15-bit since it comes directly from the clock.